File: Backend/ProjectDashboard/DTO/ProjectDTO.cs
using ProjectDashboard.Models.Domain;
using System;
using System.ComponentModel.DataAnnotations;

namespace ProjectDashboard.DTO {
	public class ProjectDTO : IDTO<Project> {

		public int Id { get; set; }

		[Required]
		public string Name { get; set; }

		[Required]
		public int TeamId { get; set; }

		[Required]
		public int OwnerId { get; set; }

		public DateTime? LastEdit { get; set; }

		[Required]
		public ContactInfo ContactPerson { get; set; }

		public void UpdateFromModel(Project item) {
			item.Name = this.Name;
			item.TeamId = this.TeamId;
			item.OwnerId = this.OwnerId;
			item.ContactPerson = this.ContactPerson;
		}
	}
}

File: Android/app/src/main/java/com/quickdev/projectdashboard/models/DTO/ProjectDTO.kt
package com.quickdev.projectdashboard.models.DTO

import com.quickdev.projectdashboard.models.domain.ContactInfo
import com.quickdev.projectdashboard.models.domain.Project
import java.time.LocalDateTime

data class ProjectDTO(
    val id: Int = 0,
    val name: String,
    val teamId: Int,
    val ownerId: Int,
    val lastEdit: LocalDateTime = LocalDateTime.now(),
    val contactPerson: ContactInfo
) : ModelDTO<Project> {

    override fun toModel(): Project {
        return Project(
            id = id,
            name = name,
            teamId = teamId,
            ownerId = ownerId,
            lastEdit = lastEdit,
            contact = contactPerson
        )
    }
}
